### ISSUE TITLE
[Core] Bugs to be fixed down the road.

### ISSUE THREAD

**Issue by AraHaan** [DecoraterBot-devs/DecoraterBot#7]
On some things there are bugs that I would like to bite after #6 is fully in place.

Changes related to this issue that I would like fixed:
- [x] Remove all command / cog specific code out of the Core.
- [x] Rework a few things on the credentials loading and the variables to remove the variables from the ``BotClient`` initialization function.
- [x] Remove the old commands list json file as that is not used anyway (and is not portable between plugins).
- [x] Convert the ``readme.md`` file to restructured text using pandoc. (The other documentation files will not work properly because of the tables they have)
- [x] Move all plugins to an separate branch or repo to use for all plugins.
- [x] Move most of the loggers to the ``plugins/logs.py`` plugin for all of the events and the game logger to ``plugins/commands.py`` plugin. This to avoid having to modify the Core for fixing the logs when an breaking discord.py change happens.
- [x] Remove ``UsedEvents.md`` file as it might not be needed much anymore since all the event usages are in an default plugin (Document it for that plugin instead).
- [x] Minimize items in the ``__init__`` function to the ``BotClient`` class.

*Some of these fixes I could probably do on my own. ~~The others I would definitely need help or a team to do it.~~*

**Comment by AraHaan** [comment]
This issue will be active to all present and future commands as well. It will be appended to when issues are found and then fixed. This would act as an central way to report bugs and fix them all in one issue.

**Comment by jakesden** [comment]
Potato.

**Comment by DavidNeon** [comment]
**Hey Ara.**
I'm at School rn so I will be helping when I can get home. Thank you

David
--------------------
NAMI CREATOR

DECORATER INSPECTOR/DEVELOPER

**Comment by AraHaan** [comment]
alright, Also if you are not ware the webhooks you made got hacked. need to figure out how to from now on encrypt them in the webhook.tk file or whatever so that way it does not expose the URL's to request resulting in lots of useless @everyone pings. (from thought of it it would be nice to have it to where webhooks cannot ping @here and @everyone.

**Comment by AraHaan** [comment]
@DavidNeon I was unsure hopw some boxes got checked when I do not have them done locally so yeah.

**Comment by AraHaan** [comment]
Alreight need to add some sort of plugin installation sysdtem that reads an json file of available plugins that can be installed and lists them out. (similar to Reads install system but different.

**Comment by AraHaan** [comment]
Alright made the json file see https://github.com/AraHaan/DecoraterBot-Plugins/blob/master/pluginlist.json

**Comment by AraHaan** [comment]
@DavidNeon You need to reclone the repo on your account.

**Comment by AraHaan** [comment]
Partially done, but need an fool proof system that would make plugin instalation never fail with an traceback when the folder to download them to does not exist.

In that reguard @jakesden mind helping me complete it so that way I can test it myself?

**Comment by AraHaan** [comment]
btw @zooba is it safe to have an 32 bit process open an 64 bit process on 64 bit windows without the 32 bit process crashing (Both command line executables that I plan to add to the path environment variable)?

**Comment by zooba** [comment]
@AraHaan Creating processes is safe - loading DLLs into memory is not.

**Comment by AraHaan** [comment]
ok, good to know now. In that case I made an change to where the user needing ffmpeg would have to clone an github repo holding the 32 bit ffmpeg exe file and then even if they are using 64- bit python it still uses the 32 bit ffmpeg. This seems to help on the fact I do not want to have to provide ffmpeg on the bot itself same for opus.

**Comment by AraHaan** [comment]
I can't do any development on this for now as my laptop has no internet right now because we got kicked out of our house earlier today because my dad was so crazy and is totally a nut.

**Comment by AraHaan** [comment]
- [ ] Move most of the loggers to the 'plugins/logs.py' plugin for all of the events and the game logger to 'plugins/commands.py' plugin. This to avoid having to modify the Core for fixing the logs when an breaking change happens.
